### PR TITLE
Update swagger to add target and credentials flag to the requests, and add examples.

### PR DESCRIPTION
**Pull request by shahzzzam**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Can you please fix the indentation so it matches the original?  The diff is extremely noisy.

**Comment by shahzzzam** [comment]
> Can you please fix the indentation so it matches the original? The diff is extremely noisy.

Yes, I am working on that

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json [hunk 1]
```diff
@@ -1337,7 +1401,7 @@
           }
         },
         "repositoryUrl": {
-          "description": "The full URL to the source code repository",
+          "description": "The full URL to the source code respository",
```
Comment (nschonni) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json]: Typo

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json [hunk 2]
```diff
@@ -1350,6 +1414,27 @@
         }
       }
     },
+    "SecretObject": {
+      "description": "Describes the properties of a secret object value.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The value of the secret. The format of this value will be determined\r\nbased on the type of the secret object. If the type is Opaque, the vaule will be\r\nused as is without any modification.",
```
Comment (nschonni) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json]: vaule -> value

**Comment by shahzzzam** [comment]
> Can you please fix the indentation so it matches the original? The diff is extremely noisy.

This has been fixed

**Comment by shahzzzam** [comment]
> Can you please fix the indentation so it matches the original? The diff is extremely noisy.

This is fixed, @jhendrixMSFT

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json [hunk 3]
```diff
@@ -788,6 +794,11 @@
             "name": "ProvisioningState",
             "modelAsString": true
           }
+        },
+        "isArchiveEnabled": {
```
[the remark this answers is not in the document]
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json]: Hi, this was a part of a PR was this property was moved from a child class to a parent class without any other modifications. I think its just how swagger decided to put them in order. We cannot do much about it here

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json [hunk 4]
```diff
@@ -1777,6 +1866,10 @@
           "description": "The Docker file path relative to the source location.",
           "type": "string"
         },
+        "target": {
+          "description": "Gets or sets the name of the target build stage for the docker build.",
```
Comment (jhendrixMSFT) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json]: The "Gets or sets" is a C#-way of describing this field and is not applicable to all languages (i.e. not all languages generate getters and setters).  Consider rewording this description (there are two other cases where this phrasing is used).

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json [hunk 5]
```diff
@@ -1060,7 +1071,7 @@
       }
     },
     "Task": {
-      "description": "The task that has the ARM resource and task properties. \r\nThe task will have all information to schedule a run against it.",
+      "description": "The task that has the ARM resource and task properties. \r\nThe  task will have all information to schedule a run against it.",
```
[the remark this answers is not in the document]
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json]: Done

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json (hunk 4 above)
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json]: done

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (shahzzzam) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json]: @nschonni can you please close this?

[the remark this answers is not in the document]
Comment (nschonni) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2018-09-01/containerregistry_build.json]: Sorry, I don't know what you'd like me to close. My comments were just some hints on minimizing the diff, but you can ignore them if they don't make sense

**Comment by shahzzzam** [comment]
@jhendrixMSFT  it seems I cannot resolve one of your reviews about indentation. Can you resolve it on your end? Seems like all the comments were addressed

**Comment by jhendrixMSFT** [comment]
@shahzzzam no worries it looks good.  Are these features live in production?

**Comment by shahzzzam** [comment]
Yes they are, just need to add them to CLI now

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**